docs/architecture/adr-002-photon-token.md

```diff
@@ -0,0 +1,216 @@
+# ADR 002: The Photon token
+
+## Changelog
+
+- 15 October 2024: Initial version
+- 30 October 2024: Revisions
+- 15 November 2024: Replace `TxFeeChecker` by `AnteDecorator`
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR proposes the introduction of the PHOTON token as the only fee token of
+AtomOne. The only way to get PHOTONs is to burn ATONEs, with a one-way burn
+that is not reversible at protocol level.
+
+The PHOTON denom is `photon`, while the base denom is `uphoton`, with:
+```
+1 photon = 1,000,000 uphoton
+```
+Any subsequent formula in this document will use the `photon` denom for
+brevity.
+
+## Context
+
+The PHOTON token is specified in the [AtomOne Constitution Article 3 Section 5]:
+
+> ### Section 5: The PHOTON Token
+>
+> The PHOTON shall be the only fee token except for ATONE to PHOTON burn
+> transactions. This applies for all transactions on the root and core shards,
+> and all IBC and ICS payments.
+>
+> ATONE tokens may be burned to PHOTON tokens at a conversion rate set by law
+> such that the total amount of PHOTONs mintable through burning ATONE tokens
+> shall be capped at 1B PHOTON tokens.
+>
+> PHOTONs cannot be converted back into ATONE tokens.
+
+## Decision
+
+The ADR proposes to create a new `x/photon` module to host the following
+features:
+- New `ConversionRate` query
+- New `MsgMintPhoton` message
+- New `AnteDecorator` implementation to enforce the PHOTON token as the only
+  fee token.
+
+### `ConversionRate` query
+
+The `ConversionRate` query returns a `sdk.Dec` which represents the current
+conversion rate of ATONE to PHOTON. This conversion rate is computed as the
+following:
+
+```math
+conversion\_rate = \dfrac{photon_{max\_supply} - photon_{supply}}{atone_{supply}}
+```
+where
+```math
+photon_{max\_supply} = 1,000,000,000
+```
+
+The `conversion_rate` will therefore be a function of the ATONE supply and the
+current PHOTON supply. The `conversion_rate` is expected to naturally decrease
+over time as the ATONE supply increases by means of inflation.
+
+### `MsgMintPhoton` message
+
+`MsgMintPhoton` takes an amount of ATONEs and returns a corresponding amount
+of PHOTONs. The amount of ATONEs is burnt while the corresponding amount of
+PHOTONs is minted and sent to the message signer account (who is also the sender
+of the ATONEs that are burnt). The number of minted PHOTONs is equal to the
+number of burnt ATONEs multiplied by the conversion rate described in the
+`ConversionRate` query section below.
+
+```math
+photons_{minted} = atones_{burned} \times conversion\_rate
+```
+
+The supply of PHOTON, although capped at 1B, will never reach this cap because
+the situation in which this happens is if all ATONEs in circulation are burnt.
+This is not possible if there are ATONEs staked, which is a requirement to be
+able to produce blocks. This also means in practice that this message is
+expected to never be able to fail because of insufficient mintable PHOTONs.
```

Comment:
```suggestion
expected to never be able to fail because of insufficient mintable PHOTONs.

There is however still a need to check that after calculations the amount of 
photons to mint is non-zero, which due to rounding is a possibility when 
burning very small fractions of the staking token (ATONE) supply.
```